### DISCUSSION TITLE
Connection timeout

### DISCUSSION THREAD

**Discussion by gstuntz** [serverpod/serverpod#2367] [Q&A] [answered]
How do I set the connection time for more than 20 Seconds for long operations.

Keep getting the below error when updating a large base64 string 

`TimeoutException after 0:00:20.000000: Future not completed
`
Kindly assist

**Comment by gstuntz** [comment, marked as answer]
Found  the solution below because of lack of documentation. 

[https://github.com/serverpod/serverpod/issues/1994#issuecomment-1981155422]

**Comment by jamesncl** [comment]
@gstuntz  I'm also trying to await a serverpod endpoint method which takes longer than 20 seconds and am running into the same error you reported, but I don't see how issue #1994 answers the question - #1994 is about increasing payload size, which isn't the problem in my case. I don't see any [configuration option in the docs](https://docs.serverpod.dev/concepts/configuration) to alter this timeout behaviour. Am I missing something?

**Comment by jamesncl** [comment]
I figured it out. The timeout is not configured on the server, it is configured on the client:

```dart
final client = Client(serverUrl, connectionTimeout: timeout)
```
